### PR TITLE
Support Marathon 1.0.0-RC1+ JSON definition for 422 error responses

### LINKED ISSUE #164: HTTP error code 422 (Unprocessable Entity) not deserialized correctly
At least for Marathon 1.1.1, go-marathon does not deserialize the error message from a 422 response properly: As found out in #162, the JSON body

```json
{
  "message": "Object is not valid",
  "details": [
    {
      "path": "\/value",
      "errors": [
        "Requested service port 10000 conflicts with a service port in app \/myapp-green2",
        "Requested service port 10000 conflicts with a service port in app \/myapp"
      ]
    }
  ]
}
```

produces the [APIError](https://github.com/gambol99/go-marathon/blob/697a616b3996092969b86950b1e555e3b4391945/error.go#L48-L52)

```
Marathon API error: Object is not valid (attribute '': )
```

We need to update the parsing logic.

### PR DESCRIPTION
**Pull request by timoreimann**
The error structure for 422 HTTP error codes has [changed again in version 1.0.0-rc1](https://github.com/mesosphere/marathon/pull/3387), adapting the same format as the 400 HTTP error response.

This is an "unwanted successor" to #133.

Fixes #164.

### CODE REVIEW

**Comment by timoreimann** [comment]
@gambol99 could you please take a look?

Apologies for the somewhat hard-to-read code but the upstream changes are really subtle and require a bit of trickery to distinguish cleanly. Suggestions for alternatives are welcome of course.

We should also consider removing support for older formats at some point.

File: error.go [hunk 1]
```diff
@@ -109,20 +109,25 @@ func (def *simpleErrDef) errCode() int {
...
 }
 
+type detailDescription struct {
+	Path   string   `json:"path"`
+	Errors []string `json:"errors"`
+}
+
+func (d detailDescription) String() string {
+	return fmt.Sprintf("path: '%s' errors: %s", d.Path,
+		strings.Join(d.Errors, ", "))
```
Comment (gambol99) [inline error.go]: nit-pick ... do we need the newline?

**Comment by gambol99** [comment]
other than the one picky comment :-) .. looks good to me ..

File: error.go (hunk 1 above)
Comment (timoreimann) [inline error.go]: Definitely not! Removed.

**Comment by timoreimann** [comment]
Thanks 👍 

Waiting for CI to go green, will merge afterwards.

**Pull request merged**